src/tasks/tasks.proj

```diff
@@ -41,7 +41,8 @@
   <Target Name="GetTasksSrc"
           DependsOnTargets="PrepareProjectReferences">
     <PropertyGroup>
-      <TasksIntermediateFile>$([MSBuild]::NormalizePath('$(ArtifactsObjDir)', '$(MSBuildProjectName)', 'Debug', 'build-semaphore.txt'))</TasksIntermediateFile>
+      <!-- if we build for several architectures in the same dirty tree, the tasks for one may be different from the tasks for another -->
+      <TasksIntermediateFile>$([MSBuild]::NormalizePath('$(ArtifactsObjDir)', '$(MSBuildProjectName)', 'Debug', 'build-semaphore-$(TargetOS)-$(TargetArchitecture).txt'))</TasksIntermediateFile>
```

Comment:
I don't think `TargetArchitecture` makes sense, the tasks that are built are conditioned based on `TargetOS` only.